classes/order/OrderHistory.php

```diff
@@ -510,12 +510,12 @@ public function sendEmail($order, $template_vars = false)
                 $data = array_merge($data, $template_vars);
             }
 
-            $data['{total_paid}'] = Tools::displayPrice((float) $order->total_paid, new Currency((int) $order->id_currency), false);
+            $context = Context::getContext();
+            $data['{total_paid}'] = Tools::getContextLocale($context)->formatPrice((float) $order->total_paid, (new Currency((int) $order->id_currency))->iso_code);
```

Comment:
Sometimes you use `new Currency`, sometimes `Currency::getCurrencyInstance` this lacks consistency
If anything I'd say `Currency::getCurrencyInstance` is lest worse because it stores the result and avoid creating too many instances, but in the end even this is not that great because it sill instanciate useless Currency objects
I think the real better way would be to add a `Currency::getIsoCodeById` method.. this would be more performant and even the could would look cleaner:

Instead of `(new Currency((int) $order->id_currency))->iso_code` -> `Currency::getIsoCodeById($order->id_currency)`

`Currency::getIsoCodeById` would just make a Db query, and maybe cache the result by id (on success) to avoid too many requests, and it musn't check for active or deleted status or this will cause bugs

Reply:
Fixed. Thanks :)